### ISSUE TITLE
Commands like show and complete should allow a filter argument

### ISSUE THREAD

**Issue by mehmetseckin** [mehmetseckin/todo-cli#13]
This will enable non-interactive use, e.g.

`todo show refactor` would show everything with refactor in it.

`todo complete registration` will complete everything with registration in it.

**Comment by Regenhardt** [comment]
For show/list, this can usually be achieved already by piping the output into grep (or the powershell equivalent) like `todo list --all | grep refactor`. I did however add a `--list` filter in #23 to the list and complete commands to only list or complete tasks from a specific list.

**Comment by mehmetseckin** [comment]
fair enough, with json output available I think this is no longer necessary.
